pkg/server/singleprocess/service_pipeline.go

```diff
@@ -390,6 +396,17 @@ func (s *Service) buildStepJobs(
 				return nil, nil, nil, err
 			}
 
+			// Add the parent step workspace ref.
+			// **Note** this only works now because we are limiting nested
+			// pipelines to 1 layer.
+			if step.Workspace != nil {
+				for _, jobReq := range embedJobs {
+					jobReq.Job.Workspace = &pb.Ref_Workspace{
+						Workspace: step.Workspace.Workspace,
```

Comment:
This makes sense, but should we also respect any nested step workspace options too? Right now this will override any workspace options set in the embedded pipeline steps like:

```hcl
pipeline "nested" {
  step "another-one" {
    workspace = "test"
    pipeline "two" {
      step "different" {
        workspace = "different"
        // ...
      }
    }
  }
}
```

I think we should preserve that step workspace if it was set explicitly, otherwise this Step Pipeline Ref could be seen as a default for any steps not explicitly set in the embedded pipeline.

Reply:
Agreed, this is the odd "bug" I mentioned in chat earlier. I've updated the logic and test code in https://github.com/hashicorp/waypoint/pull/3805/commits/ecf2e7d6259bb1d7d5f9bc04f4867522187fd476 to address this